[C++] Document connection pool vs process fork order (#100)
mongodb/docs-cpp: source/connect/connection-pools.txt

application has access to multiple threads, even if your application only
uses one thread.

Using Threads and Forks with Connection Pools
---------------------------------------------

A ``mongocxx::pool`` can be shared across multiple threads and used to create
clients. However, each ``mongocxx::client`` can only be used in a single
thread. See the :ref:`thread safety documentation <cpp-thread-safety>` for
details on how to use a ``mongocxx::client`` in a thread-safe manner.

A ``mongocxx::pool`` **cannot** be shared between a parent and a fork.

> You must create your connection pool after forking.

Reviewer: Consider creating a separate section to describe "Fork Safety" and linking to the existing [Fork Safety](https://www.mongodb.com/docs/languages/cpp/cpp-driver/current/thread-safety/#fork-safety) section. That section also notes the restriction for single-threaded `mongocxx::client`:

```
Connection Pools and Thread Safety
----------------------------------

(unchanged)

Connection Pools and Fork Safety
--------------------------------

A ``mongocxx::pool`` **cannot** be shared between processes.
You must create your connection pool after forking. See <cpp-fork-safety>.
```

> A ``mongocxx::pool`` can **not** be shared between a parent and a fork.
> You must create your connection pool after forking.

Reviewer 2: Per [term list](https://www.mongodb.com/docs/meta/style-guide/terminology/alphabetical-terms/), cannot should be one word.

```suggestion
A ``mongocxx::pool`` **cannot** be shared between a parent and a fork.
You must create your connection pool after forking.
```
